ddpui/api/pipeline_api.py

```
@@ -349,6 +349,31 @@ def cancel_queued_manual_job(request, flow_run_id, payload: TaskStateSchema):
 
         cancellation_params = {"state": payload.state.model_dump(), "force": str(payload.force)}
         res = prefect_service.cancel_queued_manual_job(flow_run_id, cancellation_params)
+
+        # Best-effort cleanup of associated Airbyte jobs
+        try:
+            task_locks = TaskLock.objects.filter(flow_run_id=flow_run_id).select_related("orgtask")
```

Ruff (0.15.12)

[error] Undefined name `TaskLock`

(F821)